pkg/machineproviders/providers/openshift/machine/v1beta1/provider.go

```diff
@@ -128,16 +170,296 @@ type openshiftMachineProvider struct {
 // - Which failure domain index does the Machine represent?
 // - Is the Machine in an error state?
 func (m *openshiftMachineProvider) GetMachineInfos(ctx context.Context, logger logr.Logger) ([]machineproviders.MachineInfo, error) {
-	return nil, nil
+	machineInfos := []machineproviders.MachineInfo{}
+
+	selector, err := metav1.LabelSelectorAsSelector(&m.machineSelector)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert label selector to selector: %w", err)
+	}
+
+	machineList := &machinev1beta1.MachineList{}
+	if err := m.client.List(ctx, machineList, &client.ListOptions{LabelSelector: selector}); err != nil {
+		return nil, fmt.Errorf("failed to list machines: %w", err)
+	}
+
+	correctFormat := isMachinesNamesFormatCorrect(machineList.Items)
+
+	for _, machine := range machineList.Items {
+		machineInfo, err := m.generateMachineInfo(machine, correctFormat, logger)
+		if err != nil {
+			return nil, err
```

Comment:
This error should be wrapped with more info
```suggestion
			return nil, fmt.Errorf("could not generate machine info for machine %s: %w", machine.Name, err)
```